themes/10up-block-theme/src/Assets.php

```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * Assets module.
+ *
+ * @package TenupBlockTheme
+ */
+
+namespace TenupBlockTheme;
+
+use TenupFramework\Module;
+use TenupFramework\ModuleInterface;
+use TenupBlockTheme\Traits\GetAssetInfo;
+
+/**
+ * Assets module.
+ *
+ * @package TenupBlockTheme
+ */
+class Assets implements ModuleInterface {
+
+	use Module;
+	use GetAssetInfo;
+
+	/**
+	 * Can this module be registered?
+	 *
+	 * @return bool
+	 */
+	public function can_register() {
+		return true;
+	}
+
+	/**
+	 * Register any hooks and filters.
+	 *
+	 * @return void
+	 */
+	public function register() {
+		add_action( 'init', [ $this, 'scripts' ] );
+		add_action( 'init', [ $this, 'register_all_icons' ], 10 );
+		add_action( 'wp_enqueue_scripts', [ $this, 'styles' ] );
+		add_action( 'enqueue_block_editor_assets', [ $this, 'editor_style_overrides' ] );
```

Comment:
Question:

Does this actually have the proper IDE integration? Based on what @tobeycodes said before this here is as bad as the `$n()` function we used to have 🤔

Reply:
It definitely _should_ work with all decent IDE',s as this is the standard way of calling class methods in PHP. The `$n()` function was an odd one as it was a wrapper around `__NAMESPACE__` which IDE's also don't seem to like. 

This is what I see in my IDE, which is what I'd expect:

<img width="912" alt="image" src="https://github.com/user-attachments/assets/7697dda8-4b7c-4ddb-849c-2e9e683a73a5" />

Reply:
Just tested and in VSCode it works also. So never mind my comment here =D